### PR TITLE
🎉 New Destination: Yellowbrick

### PR DESCRIPTION
**Pull request by markcusack**
## What
Adds a new destination connector for the Yellowbrick database following demand from Yellowbrick customers.

## How
Implements a V2 destination connector for Yellowbrick.

## Recommended reading order
1. `YellowbrickDestination.java`

## 🚨 User Impact 🚨
New destination connector. No user impact.

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

Screenshot 2024-03-02 at 5 58 20 PM

- **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- Connector version is set to `0.0.1`
    - `Dockerfile` has version `0.0.1`
- Documentation updated
    - Connector's `README.md`
    - Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - `docs/integrations/README.md`

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Connector Generator</strong></summary>

- Issue acceptance criteria met
- PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/resources/pull-requests-handbook)
- If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:generateScaffolds` then checking in your changes
- Documentation which references the generator is updated as needed

</details>

<details><summary><strong>Updating the Python CDK</strong></summary>

### Airbyter

Before merging:
- Pull Request description explains what problem it is solving
- Code change is unit tested
- Build and my-py check pass
- Smoke test the change on at least one affected connector
   - On Github: Run [this workflow](https://github.com/airbytehq/airbyte/actions/workflows/connectors_tests.yml), passing `--use-local-cdk --name=source-<connector>` as options
   - Locally: `airbyte-ci connectors --use-local-cdk --name=source-<connector> test`
- PR is reviewed and approved
      
After merging:
- [Publish the CDK](https://github.com/airbytehq/airbyte/actions/workflows/publish-cdk-command-manually.yml)
   - The CDK does not follow proper semantic versioning. Choose minor if this the change has significant user impact or is a breaking change. Choose patch otherwise.
   - Write a thoughtful changelog message so we know what was updated.
- Merge the platform PR that was auto-created for updating the Connector Builder's CDK version
   - This step is optional if the change does not affect the connector builder or declarative connectors.

</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Thanks for the contribution @markcusack I'm going to create a sandbox to run integration tests to your code.

**Comment by markcusack** [comment]
> Thanks for the contribution @markcusack I'm going to create a sandbox to run integration tests to your code.

Thanks! Is this a Yellowbrick sandbox? I.e. a connection to our free cloud-based Yellowbrick instance, or a different sandbox?

**Comment by marcosmarxm** [comment]
Free cloud-based sandbox

**Comment by markcusack** [comment]
> Free cloud-based sandbox

The Yellowbrick sandbox is fairly locked down, and regular sandbox users don't have permission to create a new database, which the integration tests require -- a database is created just for them. We can relax the permissions to allow database creation while you're running the tests, however. Let me know the email address you've signed up for and I'll ask the team to give you more extensive permissions.

**Comment by marcosmarxm** [comment]
Sent email in Slack DM.

**Comment by markcusack** [comment]
Removed a file that should not have been in the commit.

**Comment by marcosmarxm** [comment]
I triggered tests @markcusack you're now able to see results if they fail. Waiting results.

File: airbyte-integrations/connectors/destination-yellowbrick/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    cloud:
+      dockerRepository: airbyte/destination-yellowbrick
+      enabled: true
+    oss:
+      enabled: true
+  releases:
+    breakingChanges:
+      0.0.1:
+        message: >
+          This initial version implements [Destinations V2](https://docs.airbyte.com/release_notes/upgrading_to_destinations_v2/#what-is-destinations-v2), which provides better error handling, incremental delivery of data for large syncs, and improved final table structures.
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/destination-yellowbrick/metadata.yaml:25]: This is the first version. Remove the breaking change.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/destination-yellowbrick/metadata.yaml (hunk 1 above)
Comment (markcusack) [reply airbyte-integrations/connectors/destination-yellowbrick/metadata.yaml]: @marcosmarxm I've removed the releases section, since it's a new connector.

**Review by marcosmarxm — APPROVED** [review]

**Pull request merged**